server/coordinator/procedure/factory.go

```diff
@@ -39,14 +40,23 @@ type DropTableRequest struct {
 	OnFailed    func(error) error
 }
 
-func NewFactory(allocator id.Allocator, dispatch eventdispatch.Dispatch) *Factory {
+type TransferLeaderRequest struct {
+	Cluster *cluster.Cluster
+
+	ShardID           storage.ShardID
+	OldLeaderNodeName string
+	NewLeaderNodeName string
```

Comment:
No need to expose these field？

Reply:
It's used by `api.go`, these variables are need to be exposed.